### PR TITLE
Init Airdrop Schema, Clean up OP Drop 1, Init OP Token Data

### PR DESCRIPTION
**Pull request by MSilb7**
Brief comments on the purpose of your changes:

Cleaning up some naming & methodologies around OP Airdrop #1, adding basic token metadata for tracking, and initializing a generalized airdrop schema.

Note: There may be some downstream implications for how OP Airdrop 1 tables are named, they should've specified drop 1, ideally something like `airdrop.op_airdrop_1_addresses`.

Open to other schema-type modifications, since these things can happen on any chain.

Edit Feb 11 - This sprawled a bit, but also trying to unify the scattered OP-related folders within an 'OP' folder.

Side-Note: I wonder if there's some generic inputs we can set up for the `MerkleClaim` fork airdrop contracts (i.e. give the table name, field name for claimer, field name for amount, token address & that automatically populates the spell).

**For Dune Engine V2**

I've checked that:

### General checks:
* [ ] I tested the query on dune.com after compiling the model with dbt compile (compiled queries are written to the target directory)
* [ ] I used "refs" to reference other models in this repo and "sources" to reference raw or decoded tables 
* [ ] if adding a new model, I added a test
* [ ] the filename is unique and ends with .sql
* [ ] each sql file is a select statement and has only one view, table or function defined  
* [ ] column names are `lowercase_snake_cased`
* [ ] if adding a new model, I edited the dbt project YAML file with new directory path for both models and seeds (if applicable)
* [ ] if wanting to expose a model in the UI (Dune data explorer), I added a post-hook in the JINJA config to add metadata (blockchains, sector/project, name and contributor Dune usernames)

### Pricing checks:
* [ ] `coin_id` represents the ID of the coin on coinpaprika.com
* [ ] all the coins are active on coinpaprika.com (please remove inactive ones)

### Join logic:
* [ ] if joining to base table (i.e. ethereum transactions or traces), I looked to make it an inner join if possible

### Incremental logic:
* [ ] I used is_incremental & not is_incremental jinja block filters on both base tables and decoded tables
  * [ ] where block_time >= date_trunc("day", now() - interval '1 week')
* [ ] if joining to base table (i.e. ethereum transactions or traces), I applied join condition where block_time >= date_trunc("day", now() - interval '1 week')
* [ ] if joining to prices view, I applied join condition where minute >= date_trunc("day", now() - interval '1 week')

### CODE REVIEW

**Comment by MSilb7** [comment]
Gitcoin table was just decoded, so not sure what's throwing errors

File: models/airdrop/apecoin/apecoin_airdrop_transfers.sql [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    incremental_strategy = 'merge',
+    unique_key = ['transfer_tx_hash', 'transfer_evt_index', 'transfer_block_number']
+    )
+}}
+
+-- https://dune.com/queries/511739 h/t hildobby
+{% set airdrop_start_date = '2022-03-16' %}
+{% set airdrop_end_date = '2022-03-19' %}
+
+SELECT
```
Comment (augustog) [inline models/airdrop/apecoin/apecoin_airdrop_transfers.sql:15]: definitely nitpicking: missing indentation

File: models/airdrop/apecoin/apecoin_airdrop_transfers.sql [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+'ethereum' AS blockchain,
+'APE Airdrop' AS airdrop_name,
+'Apecoin' AS airdrop_project,
+
+tfer. account AS recipient_address,
```
Comment (augustog) [inline models/airdrop/apecoin/apecoin_airdrop_transfers.sql:20]: also nitpicking: space between `tfer. ` and `account`

File: models/op_token/op_token_optimism_initial_allocations.sql [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        alias='initial_allocations'
+        , unique_key = ['allocation_category','allocation_subcategory']
+        , post_hook='{{ expose_spells(\'["optimism"]\',
+                                  "project",
+                                  "ovm_optimism",
+                                  \'["msilb7"]\') }}'
+  )
+}}
+
+, initial_allocation_percentages AS (
```
Comment (augustog) [inline models/op_token/op_token_optimism_initial_allocations.sql:14]: this is the reason why the slim ci is failing. seems you are missing a CTE?

**Review by augustog — CHANGES_REQUESTED** [review]
I had a few minor (nitpicky) comments about style, but, importantly, seems like one of the models is missing one or more CTEs (as it starts with `, initial_allocation_percentages as (` with no `with` clause

File: models/op_token/op_token_optimism_initial_allocations.sql (hunk 3 above)
Comment (MSilb7) [reply models/op_token/op_token_optimism_initial_allocations.sql]: copy and paste error. resolved

File: models/airdrop/apecoin/apecoin_airdrop_transfers.sql (hunk 1 above)
Comment (MSilb7) [reply models/airdrop/apecoin/apecoin_airdrop_transfers.sql:15]: where should the indentation be? The start of the line?

**Comment by MSilb7** [comment]
* Adding OP Airdrop 2, then this will be ready again

**Comment by MSilb7** [comment]
Info is in, but getting errors that the address list for drop #2 is too big.

**Comment by MSilb7** [comment]
Still getting a "query too large" error. Maybe it should be a csv upload?

**Comment by Hosuke** [comment]
> Still getting a "query too large" error. Maybe it should be a csv upload?

Probably yes.

**Comment by MSilb7** [comment]
removing the airdrop 2 address list for now, we'll likely just csv upload it.

**Comment by MSilb7** [comment]
found some token joins that caused results to sprawl to a ton of rows. Should be fixed now.

File: models/labels/airdrop/persona/optimism_rounds/labels_airdrop_1_receivers_optimism.sql [hunk 4]
```diff
@@ -3,12 +3,12 @@
...
     'airdrop' AS category,
     'soispoke' AS contributor,
     'query' AS source,
     timestamp('2022-09-29') as created_at,
     now() as updated_at,
     'op_airdrop_1_receivers' AS model_name,
     'persona' as label_type
-FROM {{ ref('airdrop_optimism_addresses_1') }}
\ No newline at end of file
+FROM {{ ref('airdrop_optimism_addresses_1') }}
```
Comment (Hosuke) [inline models/labels/airdrop/persona/optimism_rounds/labels_airdrop_1_receivers_optimism.sql]: Hi @MSilb7, please review the `labels` part to check if I make any mistake during resolving conflicts.

Comment (MSilb7) [reply models/labels/airdrop/persona/optimism_rounds/labels_airdrop_1_receivers_optimism.sql]: hey seems good. It's just this ref name right"

Comment (Hosuke) [reply models/labels/airdrop/persona/optimism_rounds/labels_airdrop_1_receivers_optimism.sql]: yes

**Comment by MSilb7** [comment]
Looks like theres some merge conflicts. There may be files that accidentally got pulled over from other PRs ( token factory, aave)

**Comment by jeff-dude** [comment]
> Looks like theres some merge conflicts. There may be files that accidentally got pulled over from other PRs ( token factory, aave)

please do work through the merge conflicts and push when ready

**Comment by MSilb7** [comment]
> > Looks like theres some merge conflicts. There may be files that accidentally got pulled over from other PRs ( token factory, aave)
> 
> please do work through the merge conflicts and push when ready

I'm trying to figure out how to address these. The button is grayed out for me, and the conflicts aren't showing for me in vscode

**Comment by MSilb7** [comment]
bumping this + there's also the airdrop #2 address list which was too large to put in a table: https://github.com/ethereum-optimism/op-analytics/blob/main/reference_data/address_lists/op_airdrop2_addresses.csv

**Review by Hosuke — APPROVED** [review]
LGTM.

[Sample airdrop transfers](https://dune.com/queries/2221903)
[Sample op_token_optimism.inflation_schedule](https://dune.com/queries/2221912) (single line)
[Sample ovm_optimism_l2_token_factory](https://dune.com/queries/2221925)
[Sample op_token_metadata](https://dune.com/queries/2221943) (single line)
[Sample op_token_optimism_initial_allocations](https://dune.com/queries/2221952)
[Sample ovm_optimism_cross_domain_messenger_messages](https://dune.com/queries/2221957)

**Comment by jeff-dude** [comment]
> bumping this + there's also the op airdrop #2 address list which was too large to put in a table: https://github.com/ethereum-optimism/op-analytics/blob/main/reference_data/address_lists/op_airdrop2_addresses.csv

as for large files, are you in contact with niles and ingest team for their upload process? that may be best, cc @nileslawrence

**Comment by jeff-dude** [comment]
@MSilb7 can you help me understand the intention of this airdrop spell better?
- i'm seeing them set up as incremental, but most have an end date. if there is an end date, what is the use case for incremental ongoing loads? 
- some models have end date variable, but no usage downstream
- some models don't have end date variables, assuming that means claim window is still open?

i wonder if these all make sense to be tables, rather than incremental? and if there is an end date, a `static` tag can be applied in config to help our orchestration know that it doesn't need run on an ongoing basis?

i also thought maybe the intention was to be tracking the airdrop transfers over time, but the inner join to `transactions` base table on `block_number` and particular hash make me think it isn't ongoing.

**Comment by jeff-dude** [comment]
closing this, can track in tagged PRs

**Pull request closed without merging**